### ISSUE TITLE
Salt&Pepper attack should return minimum-norm solutions

### ISSUE THREAD

**Issue by maurapintor** [bethgelab/foolbox#619]
Salt and Pepper attack should compute the norm of the perturbation, instead of the one of the perturbed sample.

```python
norms = flatten(x).norms.l2(axis=-1) 
```

Should be probably changed to:

```python
norms = flatten(x - x0).norms.l2(axis=-1) 
```

https://github.com/bethgelab/foolbox/blob/2343d589fb80e0e8931827ac58cadc83e6e2f4d2/foolbox/attacks/saltandpepper.py#L92

**Comment by jonasrauber** [comment]
Uh, this one is bad. Thanks @maurapintor! This bug was introduced when porting the attacks to Foolbox 3 and we should have tested more carefully.

**Comment by jonasrauber** [comment]
Thanks @zangobot for PR #620 to fix this.

**Comment by jonasrauber** [comment]
I have released Foolbox 3.3.1 with the bug fix: https://github.com/bethgelab/foolbox/releases/tag/v3.3.1

@maurapintor Do you now get better results?

**Comment by maurapintor** [comment]
> I have released Foolbox 3.3.1 with the bug fix: https://github.com/bethgelab/foolbox/releases/tag/v3.3.1
> 
> @maurapintor Do you now get better results?

Yes, thanks! Here is a 2D example, the green line represents the queries to the model, the red hexagon is the starting point and the green start the last query.  Now it returns the correct point (minimum distance and adversarial).

![image](https://user-images.githubusercontent.com/23276849/108813637-b3bb7480-75b1-11eb-8270-e141819ffc5c.png)

While the previous version was returning this:

![image](https://user-images.githubusercontent.com/23276849/108813881-20cf0a00-75b2-11eb-89cc-092cc74232b3.png)

So basically the point with minimum distance from the origin. Now it's fixed :)
